src/script/interpreter.cpp

```diff
@@ -1776,6 +1776,85 @@ bool SignatureHashBIP143(uint256 &sighashOut, const CScript &scriptCode,
     return true;
 }
 
+static const CHashWriter HASHER_TAPSIGHASH = TaggedHash("TapSighash");
+
+template <class T>
+bool SignatureHashBIP341(uint256 &hash_out,
+                         const std::optional<ScriptExecutionData> &execdata,
+                         const T &tx_to, uint32_t in_pos,
+                         SigHashType sig_hash_type,
+                         const PrecomputedTransactionData &cache) {
+    uint8_t ext_flag = bool(execdata);
+    uint8_t key_version = 0;
+    assert(in_pos < tx_to.vin.size());
+
+    CHashWriter ss = HASHER_TAPSIGHASH;
```

Comment:
This creates a copy, right?

Reply:
It has to, GetSHA256 invalidates the object at the end of the function